packages/scandipwa/src/component/Menu/Menu.component.tsx

```diff
@@ -183,24 +183,35 @@ export class MenuComponent extends PureComponent<MenuComponentProps> {
 
     renderSubMenuDesktopItems(item: FormattedMenuItem): ReactElement {
         const { item_id, children } = item;
+        const { visitedMenuItems, updateVisitedMenuItems, removeVisitedMenuItems } = this.props;
 
         if (!Object.keys(children).length) {
             return null;
         }
 
         const { activeMenuItemsStack, closeMenu } = this.props;
         const isVisible = activeMenuItemsStack.includes(item_id);
+        const isExpandable = visitedMenuItems.length === 0;
+        const isCollapsed = !isVisible && visitedMenuItems.includes(item_id) && !activeMenuItemsStack.length;
+
+        if (isCollapsed) {
+            removeVisitedMenuItems(item_id);
```

Comment:
as mentioned previously, there shouldn't be any function calls in render methods. Have you tried refactoring this implementation?